src/interface/variable.hpp

```diff
@@ -157,6 +164,13 @@ class CellVariable {
   const int sparse_id_;
   const std::array<int, 6> dims_, coarse_dims_;
 
+  // Machinery for giving each variable a unique ID that is faster to
+  // evaluate than a string. Assumes number of MPI ranks unchanged
```

Comment:
Does this mean the sim cannot be restarted with a different number of MPI ranks?

Reply:
I am not sure what I meant when I wrote this comment. The basic idea here is a unique hash where variables are named in order of creation. I think that should be deterministically set at startup and not dependent on MPI rank number.

Let me double check it works with a restart with different # of MPI ranks though.

Reply:
Ok confirmed. Here's the final value of one of the fields of sparse advection, started on 2 ranks and restarted on 4.
![0020](https://user-images.githubusercontent.com/1691313/220792980-87231226-d1f6-4781-a2b4-a9d913c03233.png)